File: app/src/main/java/com/samhalperin/cooperhewitt/detailscreen/DetailContract.java
package com.samhalperin.cooperhewitt.detailscreen;

import com.samhalperin.cooperhewitt.data.models.detailobject.DetailObject;

public interface DetailContract {

    public interface View {
        void displayDetailObject(DetailObject detailObject);
        void shareDetailObject(DetailObject detailObject);
    }

    public interface UserActionsHandler {
        void loadDetailObject(String id);
        void shareDetailObject(String id);
    }
}

File: app/src/main/java/com/samhalperin/cooperhewitt/detailscreen/DetailContract.kt
package com.samhalperin.cooperhewitt.detailscreen

import com.samhalperin.cooperhewitt.data.models.detailobject.DetailObject

interface DetailContract {

    interface View {
        fun displayDetailObject(detailObject: DetailObject)
        fun shareDetailObject(detailObject: DetailObject)
    }

    interface UserActionsHandler {
        fun loadDetailObject(id: String)
        fun shareDetailObject(id: String)
    }
}
